cpp/src/glm/qn/qn_solvers.cuh

```diff
@@ -139,10 +139,16 @@ inline OPT_RETCODE min_lbfgs(const LBFGSParam<T> &param,
     LINE_SEARCH_RETCODE lsret =
       ls_backtrack(param, f, fx, x, grad, step, drt, xp, dev_scalar, stream);
 
-    bool isLsSuccess = lsret == LS_SUCCESS;
-    CUML_LOG_TRACE("Iteration %d, fx=%f", *k, fx);
-
-    if (!isLsSuccess || isnan(fx) || isinf(fx)) {
+    bool isLsValid = !isnan(fx) && !isinf(fx);
```

Comment:
Nitpick: can we put this logic in a function that both min_lbfgs and min_owlqn can share?